src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -653,12 +654,21 @@ await _projectSystemProjectFactory.ApplyBatchChangeToWorkspaceMaybeAsync(useAsyn
                         newSolution: solutionChanges.Solution.RemoveProjectReference(Id, projectReference));
                 }
 
+                // Analyzer reference removing...
+                {
+                    projectUpdateState = projectUpdateState.WithIncrementalAnalyzerReferencesRemoved(_analyzersRemovedInBatch);
+
+                    foreach (var analyzerReference in _analyzersRemovedInBatch)
+                        solutionChanges.UpdateSolutionForProjectAction(Id, solutionChanges.Solution.RemoveAnalyzerReference(Id, analyzerReference));
+                }
+
                 // Analyzer reference adding...
-                solutionChanges.UpdateSolutionForProjectAction(Id, solutionChanges.Solution.AddAnalyzerReferences(Id, _analyzersAddedInBatch));
+                {
+                    projectUpdateState = projectUpdateState.WithIncrementalAnalyzerReferencesAdded(_analyzersAddedInBatch);
 
-                // Analyzer reference removing...
-                foreach (var analyzerReference in _analyzersRemovedInBatch)
-                    solutionChanges.UpdateSolutionForProjectAction(Id, solutionChanges.Solution.RemoveAnalyzerReference(Id, analyzerReference));
+                    solutionChanges.UpdateSolutionForProjectAction(
+                        Id, solutionChanges.Solution.AddAnalyzerReferences(Id, projectUpdateState.AddedAnalyzerReferences));
```

Comment:
>projectUpdateState.AddedAnalyzerReferences) [](http://example.com/codeflow?start=79&length=43)

nit: this is also just _analyzersAddedInBatch, right? Seems a bit odd that the removing one uses _analyzersRemovedInBatch for it's solution update, whereas the adding on uses projectUpdateState.AddedAnalyzerReferences

Reply:
ick.  yes.  i will make consistent.